File: core/src/main/java/forpleuvoir/dhwu1a/core/event/request/RequestEventType.java
package forpleuvoir.dhwu1a.core.event.request;

/**
 * 申请事件类型
 *
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.event.request
 * <p>#class_name RequestEventType
 * <p>#create_time 2021/7/4 1:43
 */
public enum RequestEventType {
    NewFriendRequestEvent(NewFriendRequestEvent.class),
    MemberJoinRequestEvent(MemberJoinRequestEvent.class),
    BotInvitedJoinGroupRequestEvent(BotInvitedJoinGroupRequestEvent.class);


    private final Class<? extends RequestEvent> clazz;

    RequestEventType(Class<? extends RequestEvent> clazz) {
        this.clazz = clazz;
    }

    public Class<? extends RequestEvent> getClazz() {
        return clazz;
    }


    public static boolean hasKey(String key) {
        var isHas = false;
        for (RequestEventType value : RequestEventType.values()) {
            isHas = value.name().equals(key);
            if (isHas) break;
        }
        return isHas;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/event/request/RequestEventType.kt
package forpleuvoir.dhwu1a.core.event.request

/**
 * 申请事件类型
 *
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.event.request
 *
 * #class_name RequestEventType
 *
 * #create_time 2021/7/4 1:43
 */
enum class RequestEventType(val clazz: Class<out RequestEvent?>) {
    NewFriendRequestEvent(forpleuvoir.dhwu1a.core.event.request.NewFriendRequestEvent::class.java),
    MemberJoinRequestEvent(forpleuvoir.dhwu1a.core.event.request.MemberJoinRequestEvent::class.java),
    BotInvitedJoinGroupRequestEvent(forpleuvoir.dhwu1a.core.event.request.BotInvitedJoinGroupRequestEvent::class.java);

    companion object {
        fun hasKey(key: String): Boolean {
            var isHas = false
            for (value in values()) {
                isHas = value.name == key
                if (isHas) break
            }
            return isHas
        }
    }
}